qiskit/transpiler/passes/layout/disjoint_utils.py

```diff
@@ -111,7 +111,10 @@ def split_barriers(dag: DAGCircuit):
             barrier_uuid = f"{node.op.label}_uuid={uuid.uuid4()}"
         else:
             barrier_uuid = f"_none_uuid={uuid.uuid4()}"
-        split_dag = DAGCircuit()
+        split_dag = DAGCircuit(
+            _node_count_hint=3 * num_qubits,
+            _edge_count_hint=2 * num_qubits,
```

Comment:
Usual question for me, how did you reach a node count (hint) of `3*num_qubits` here?

Reply:
The dagcircuit being built here is taking a n qubit barrier and splitting it into n 1 qubit barriers so we can compute the connected components of the circuit. So if it's a 3q barrier it would looks something like:

![dag_barrier](https://github.com/user-attachments/assets/8efe6297-a157-47a5-9173-7daf9b5a4eda)

So there are 3 nodes for every qubit and 2 edges.